webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/v2/getAllResourcesInProjectPrequeryStandard.scala.txt

```diff
@@ -0,0 +1,46 @@
+@*
+ * Copyright © 2015-2021 the contributors (see Contributors.md).
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ *@
+
+@import org.knora.webapi._
+@import org.knora.webapi.messages.SmartIri
+
+@*
+ * Constructs a prequery that gets all resources from the specified project using standard SPARQL.
+ * This template is used only by getAllResourcesInProjectPrequery.sparql.txt.
+ *
+ * @param projectIri        the IRI of the project.
+ *@
+@(projectIri: IRI)
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+
+SELECT DISTINCT ?resource
+WHERE {
+    ?resource knora-base:attachedToProject <@projectIri> .
+    ?resourceType rdfs:subClassOf* knora-base:Resource .
+    ?resource rdf:type ?resourceType .
+    ?resource knora-base:creationDate ?creationDate.
+
+    FILTER NOT EXISTS {
+        ?resource knora-base:isDeleted true .
```

Comment:
Suppose this happens:

1. Resource A is created.
2. Resource B links to resource A.
3. Resource A is marked as deleted.

Now `getProjectResourcesWithHistoryV2` is run. It will not find resource A, because it is marked as deleted. So it will not generate an event for (1), but it will generate an event for (2). But then won't the event for (2) fail because resource A doesn't exist?

Unless I've misunderstood, I think that the events will have to create resource A and then mark it as deleted.

Reply:
Ah sorry, you are right, I should remove this `Filter Not Exists` block.